provider/cloudflare/cloudflare.go

```diff
@@ -179,17 +200,65 @@ func (z zoneService) GetZone(ctx context.Context, zoneID string) (*zones.Zone, e
 	return z.service.Zones.Get(ctx, zones.ZoneGetParams{ZoneID: cloudflare.F(zoneID)})
 }
 
-func (z zoneService) CustomHostnames(ctx context.Context, zoneID string, page int, filter cloudflarev0.CustomHostname) ([]cloudflarev0.CustomHostname, cloudflarev0.ResultInfo, error) {
-	return z.serviceV0.CustomHostnames(ctx, zoneID, page, filter)
+func (z zoneService) CustomHostnames(ctx context.Context, zoneID string) autoPager[custom_hostnames.CustomHostnameListResponse] {
+	params := custom_hostnames.CustomHostnameListParams{
+		ZoneID: cloudflare.F(zoneID),
+	}
+	return z.service.CustomHostnames.ListAutoPaging(ctx, params)
+}
+
+// listAllCustomHostnames extracts all custom hostnames from the iterator
+func listAllCustomHostnames(iter autoPager[custom_hostnames.CustomHostnameListResponse]) ([]CustomHostname, error) {
+	var customHostnames []CustomHostname
+	for ch := range autoPagerIterator(iter) {
+		customHostnames = append(customHostnames, CustomHostname{
+			ID:                 ch.ID,
+			Hostname:           ch.Hostname,
+			CustomOriginServer: ch.CustomOriginServer,
+			CustomOriginSNI:    ch.CustomOriginSNI,
+		})
+	}
+	if iter.Err() != nil {
+		return nil, iter.Err()
+	}
+	return customHostnames, nil
 }
 
 func (z zoneService) DeleteCustomHostname(ctx context.Context, customHostnameID string, params custom_hostnames.CustomHostnameDeleteParams) error {
 	_, err := z.service.CustomHostnames.Delete(ctx, customHostnameID, params)
 	return err
 }
 
-func (z zoneService) CreateCustomHostname(ctx context.Context, zoneID string, ch cloudflarev0.CustomHostname) (*cloudflarev0.CustomHostnameResponse, error) {
-	return z.serviceV0.CreateCustomHostname(ctx, zoneID, ch)
+func (z zoneService) CreateCustomHostname(ctx context.Context, zoneID string, ch CustomHostname) error {
+	params := custom_hostnames.CustomHostnameNewParams{
+		ZoneID:   cloudflare.F(zoneID),
+		Hostname: cloudflare.F(ch.Hostname),
+	}
+
+	if ch.SSL != nil {
```

Comment:
This smells like extract function refactoring. Put the whole branch in a function please